docs/enhancements/backup-restore-support.md

```diff
@@ -0,0 +1,152 @@
+  * Extract BMH's status and add `baremetalhost.metal3.io/status` to the object metadata.
+  * Add also, add the `baremetalhost.metal3.io/paused` annotation:
+    * This is required since once the annotation is added, BMO updates the BMH's status and delete the annotation.
+    * The [paused](https://github.com/metal3-io/baremetal-operator/blob/main/docs/api.md#pausing-reconciliation) annotation, is similar to the `detached` annotation we're already using - but, it's also pausing the reconcile loop. Thus, keeping the status annotation intact.
+* If a BMH is missing 'Status' (and, has `status`+`paused` annotation):
+  * To restore the original Status:
+    * Remove the `baremetalhost.metal3.io/paused` annotation
+      * Status is restored, but `baremetalhost.metal3.io/status` annotation is removed.
+      * The `status` and `paused` annotation will be restored (see previous bullet). 
+
+#### Solution for deleted Agent CRs
+
+1. Don't delete the Agent CR when an associated host doesn't exits on DB.
+   * Remove relevant [code](https://github.com/openshift/assisted-service/blob/5bbf696fc0cc5a220fd8a0d1e408ab0bcc889e3f/internal/controller/controllers/agent_controller.go#L143-L151) from AgentController.
+
+2. Restore the Host record in DB according to properties from the Agent CR.
+   * Seems we can infer more than the mandatory properties:
+     * host.ID <-- agent.Name
+     * host.Role <-- agent.Role
+     * host.InfraEnvID <-- agent.metadata.annotations['infraenvs.agent-install.openshift.io']
+     * host.Kind <-- 'Host' or 'AddToExistingClusterHost' if cluster is already installed
+     * host.InstallerArgs <-- agent.InstallerArgs
+     * host.RequestedHostname <-- agent.Hostname
+   * The entire inventory exists in Agent.Status. But, need to ensure it exists also post restore. If not, we can consider adding an annotation for it (i.e something as `status` in BMH).
+   * Note: some of the inventory data can also be restored from existing annotations.
+     E.g.:
+     * inventory.agent-install.openshift.io/cpu-architecture: x86_64
+     * inventory.agent-install.openshift.io/cpu-virtenabled: "true"
+     * inventory.agent-install.openshift.io/host-isvirtual: "true"
+     * inventory.agent-install.openshift.io/host-manufacturer: RedHat
+     * inventory.agent-install.openshift.io/host-productname: KVM
+     * inventory.agent-install.openshift.io/storage-hasnonrotationaldisk: "false"
+   
+### Risks and Mitigations
+
+* In general, these changes might affect flows on spoke cluster (before and after restore), need to ensure no regressions on all scenarios.
+* Adding the 'paused' annotation means a stale BMH status, need to check if it's crucial.
+* Keeping the Agent CR and restoring values into Host in DB could affect current flows (e.g. BindHost, UnbindHost, etc).
+* Restoring Hosts will result in partial records in DB. E.g. missing Inventory - could be challenging to tackle (I.e. probably need to add annotations to Agent).
```

Comment:
I don't think we'd be missing inventory, but do we know of any other gaps here?

Reply:
Not yet, would need to investigate.